hooks/command

```diff
@@ -212,6 +212,13 @@ while IFS='=' read -r name _ ; do
   fi
 done < <(env | sort)
 
+# Parse host mappings and add them to the docker args
+while IFS='=' read -r name _ ; do
```

Comment:
This isn't really a question related to this PR, but I wanted to ask @lox: is there any reason we prefer this syntax, over a simpler for loop?

```shell
i=0
while true; do
  env_name="BUILDKITE_PLUGIN_DOCKER_ADD_HOST_$i"
  if [[ -z "${!env_name:-}" ]]; then break; fi
  args+=( "--add-host" "${!env_name}" )
  i=$((i+1))
done
```

Reply:
I use this pattern when I want to refer to the $i later on.

Reply:
Oh, sorry, I see what you mean.

Reply:
I tend to use the while read loop for most things because I understand it's edge cases really well and usually I am iterating over output from another command and this is the only way to set arguments in the scope outside of the while loop.

In this example, I'd be totally 👌🏻 for a simpler loop.

Reply:
Cool! That makes total sense @lox!